src/basic/fileio.c

```diff
@@ -856,35 +857,41 @@ int read_full_file_full(
         return read_full_stream_full(f, filename, offset, size, flags, ret_contents, ret_size);
 }
 
-int executable_is_script(const char *path, char **interpreter) {
-        _cleanup_free_ char *line = NULL;
-        size_t len;
-        char *ans;
+int script_get_interpreter(const char *path, char **ret) {
```

Comment:
hmm, ELF has a concept of an interpreter. Which is very similar. hence I#d avoid the word.

maybe be be explicit that this is about shebangs. i.e. `script_get_shebang()`?

Reply:
i guess you have a point. but then again maybe go for script_get_shebang_interpreter() is the best of all worlds?